app/models/announcement.rb

```diff
@@ -49,7 +50,9 @@ def mentions
   end
 
   def statuses
-    @statuses ||= Status.from_text(text)
+    return [] if status_ids.nil?
```

Comment:
I think I would've opted for a `begin`/`end` block with `if`/`else` inside the assignment instead